### DISCUSSION TITLE
PDF 文件预览问题

### DISCUSSION THREAD

**Discussion by wuxianucw** [hydro-dev/Hydro#125] [General]
个人感觉这个问题还是后端的 MIME 发错了，应为 `application/pdf` 实际上发送了 `binary/octet-stream`，所以会唤起下载。对于一般的附加文件，使用这个 MIME 非常正确，但是对于需要预览的 PDF 文件，这个 MIME 会导致难以使用 `iframe`（唤起下载）或 `object`（MIME 对不上）来内联到题面中。

这里预想了几种解决方案：

- 后端对 PDF 文件发送 `application/pdf` 的 MIME；
- 使用插件方式引入 PDF.js 尝试解决，但目前尚不清楚对于 `binary/octet-stream` 该库能否正常工作。

**Comment by wuxianucw** [comment]
PDF.js 可以解决，该库对于 `binary/octet-stream` 仍然正常工作。开始着手编写 PDF 插件。

**Comment by wuxianucw** [comment]
[hydro-pdf-preview](https://github.com/Ri-moe/hydro-pdf) 插件已完成，使用该插件插入 `hpdf` 即可。

例：

```
@[hpdf](file://1.pdf)
```

支持任意路径，如果跨域则需要设置正确。

**Reply by hello8693** [reply]
可是codeforces的题改了之后有自己变回去了怎么办
